students/john_hunter/lesson03/mailroom.py

```diff
@@ -0,0 +1,191 @@
+
+donors = [['Frank Merriweather', 10, 15, 100], ['Thomas Tran', 5, 17, 23], \
+          ['Stephanie Terrance', 31, 48, 108], ['Sam Robidas', 4, 90, 101], \
+          ['Sandy Cohen', 29, 41, 70], ['Shioban Kemp', 2, 23000, 19]]
+
+def user_selection():
+    """Basic UI to prompt user and handle selections
+
+    Returns: 'entry' a string with value 1, 2, or 3 selected by the user
+    """
+    options = ('Send a Thank You', 'Create a Report', 'quit')
+    print('Enter \'exit\' at anytime to quit')
+    print('Select the Menu Option by Number')#Menu Selection Direction Output
+    for item in options:##Prints the options for the user to select
+        index = options.index(item)
+        print(str(index+1)+ '. ' + item)
+    entry = input('enter option by number:')
+    if entry == 'exit':
+        quit_it()
+    if not entry.isnumeric() or int(entry) < 1 or int(entry) > 3:
+        print('You must select a number 1, 2, or 3.')
+        user_selection()
+    else:
+        print('You have selected ' + options[int(entry)-1])
+    return entry
+
+def send_ty():
+    """Returns the text of the email with the donor name, description of the
+    donations
+    Prompts to enter a 'Full Name' or list, if the name is not part of the
+    list, then add that name to the list of donors and use the name
+    Prompt for a donation amount, type as a number and enter into the dictionary
+    Generate the Email thanking the donor, print it to terminal
+    return to the prompt
+    """
+    names = list()
+    #Format a list of the Donor names
+    max_len_of_name = 0
+
+    for i in range(1, len(donors)):
+        if len(donors[i][0]) > max_len_of_name:
+            max_len_of_name = len(donors[i][0])
+
+    while True:
+        input_name = input('Enter full name of donor or \'list\' to request list of donors: ')
+        for j in range(len(donors)):
+            names.append(donors[j][0])
+
+        if input_name == 'list':
+            for k in range(len(donors)):
+                name = donors[k][0]
+                print('{:{align}{width}}'.format(name, align='<', width=max_len_of_name))
+            else:
+                continue
+
+        if input_name == 'exit':
+            quit_it()
+
+        if input_name in names:
+            list_or_not = 'y'
+            break
+
+        else:
+            list_or_not = 'n'
+        if list_or_not == 'n':
+            add_donor(input_name)
+            list_or_not = 'n'
+            more_donations = 'n'
+            break
+    mas_donations = 'Would you like to add more donations for ' + input_name + '? y/n: '
+    if list_or_not == 'y':
+        more_donations = input(mas_donations)
+    if more_donations == 'y':
+        totals = add_donations(input_name)
+    else:
+        totals = 0
+        for z in range(len(donors)):
+            if input_name == donors[z][0]:
+                for c in range(len(donors[z])-1):
+                    totals = totals + int(donors[z][c+1])
+                break
```

Comment:
no need to use range() here and no need to loop to get totals.

utilize slice syntax to get list of donations, then use sum() function